store/sqlstore/post_store.go

```diff
@@ -1684,41 +1684,36 @@ func (s *SqlPostStore) buildSearchUserFilterClause(users []string, paramPrefix s
 	clause := strings.Join(clauseSlice, ", ")
```

Comment:
This needs to be changed to the actual args as well. This still uses the old style of ":paramName" to bind arguments.

Overall, a lot of this code needs to be rewritten to use squirrel completely and not rely string operations. Which means `buildSearchPostFilterClause`, `buildCreateDateFilterClause` need to be converted as well.

We should move out of `map[string]interface{}` completely.

Sorry, I should have clarified this earlier in the issue itself. This is a relatively big task. Let's take our time and do it properly.